ref_builder/repo.py

```diff
@@ -242,23 +242,45 @@ def use_transaction(self) -> Generator[Transaction, None, None]:
         if self._transaction:
             raise TransactionExistsError
 
+        if self.last_id != self.head_id:
+            logger.error(
+                "Head ID and last event ID do not match.",
+                head_id=self.head_id,
+                last_id=self.last_id,
+            )
+
+            raise TransactionExistsError
+
         if not self._lock.locked:
             raise LockRequiredError
 
+        self.prune()
```

Comment:
If line 257 is executed, it will never have an effect on the repository because you already verified the repository is consitent at line 245.

The call to `self.prune()` should go before the check at line 245. The check is a failsafe in case prune does not work as expected.